{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#packages\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sn\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import the two dataframes\n",
    "df1 = pd.read_csv(\"../input/Train_call.txt\", delimiter=\"\\t\")\n",
    "df2 = pd.read_csv(\"../input/Train_clinical.txt\", delimiter=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Nclone</th>\n",
       "      <th>Array.129</th>\n",
       "      <th>Array.34</th>\n",
       "      <th>Array.67</th>\n",
       "      <th>Array.24</th>\n",
       "      <th>Array.22</th>\n",
       "      <th>Array.36</th>\n",
       "      <th>...</th>\n",
       "      <th>Array.64</th>\n",
       "      <th>Array.89</th>\n",
       "      <th>Array.30</th>\n",
       "      <th>Array.35</th>\n",
       "      <th>Array.93</th>\n",
       "      <th>Array.10</th>\n",
       "      <th>Array.123</th>\n",
       "      <th>Array.100</th>\n",
       "      <th>Array.134</th>\n",
       "      <th>Array.130</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>6</td>\n",
       "      <td>156012158</td>\n",
       "      <td>157939123</td>\n",
       "      <td>85</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>4</td>\n",
       "      <td>119819380</td>\n",
       "      <td>119910398</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>9</td>\n",
       "      <td>1330131</td>\n",
       "      <td>5289937</td>\n",
       "      <td>382</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1990</th>\n",
       "      <td>16</td>\n",
       "      <td>46232996</td>\n",
       "      <td>46704264</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>924</th>\n",
       "      <td>7</td>\n",
       "      <td>38163197</td>\n",
       "      <td>38204539</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>3</td>\n",
       "      <td>16617887</td>\n",
       "      <td>16778176</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2559</th>\n",
       "      <td>20</td>\n",
       "      <td>12278907</td>\n",
       "      <td>12478772</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>745</th>\n",
       "      <td>6</td>\n",
       "      <td>20726739</td>\n",
       "      <td>22048197</td>\n",
       "      <td>107</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>2</td>\n",
       "      <td>23809995</td>\n",
       "      <td>25541135</td>\n",
       "      <td>190</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>4</td>\n",
       "      <td>31107518</td>\n",
       "      <td>36697482</td>\n",
       "      <td>208</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows Ã— 104 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome      Start        End  Nclone  Array.129  Array.34  Array.67  \\\n",
       "882            6  156012158  157939123      85         -1        -1        -1   \n",
       "581            4  119819380  119910398       6          0         0        -1   \n",
       "1201           9    1330131    5289937     382         -1        -1         0   \n",
       "1990          16   46232996   46704264      30          1         0        -1   \n",
       "924            7   38163197   38204539       6          1         1         0   \n",
       "385            3   16617887   16778176       9          0         0         0   \n",
       "2559          20   12278907   12478772      12          0         0         0   \n",
       "745            6   20726739   22048197     107          0         0         0   \n",
       "287            2   23809995   25541135     190          1         0         0   \n",
       "512            4   31107518   36697482     208         -1         0         0   \n",
       "\n",
       "      Array.24  Array.22  Array.36    ...      Array.64  Array.89  Array.30  \\\n",
       "882         -1        -1         0    ...             0         0         0   \n",
       "581          0        -1         0    ...             0         0         0   \n",
       "1201         1         0         0    ...             1         0         0   \n",
       "1990         0        -1         0    ...             0         0         0   \n",
       "924          1         0         0    ...             1         0         0   \n",
       "385          1         1         0    ...             0         0         1   \n",
       "2559         0         0         1    ...            -1         0         0   \n",
       "745          0         1         0    ...             1         0         1   \n",
       "287         -1         0         0    ...            -1         0         0   \n",
       "512         -1         0         0    ...            -1         0         0   \n",
       "\n",
       "      Array.35  Array.93  Array.10  Array.123  Array.100  Array.134  Array.130  \n",
       "882         -1         1        -1         -1          0         -1          1  \n",
       "581          0         0        -1          0          0          0          0  \n",
       "1201         0         2         0         -1          0         -1          0  \n",
       "1990         0         0         0          0          0         -1          0  \n",
       "924          0         1         0         -1          0          1          0  \n",
       "385          0        -1         0          0          0          1          0  \n",
       "2559         0         0         0         -1          0          1         -1  \n",
       "745          0         1         0          1          0          1          0  \n",
       "287          0         1         0         -1          0          0          0  \n",
       "512          0         0        -1          0          0          0         -1  \n",
       "\n",
       "[10 rows x 104 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#explore df1\n",
    "df1.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Nclone</th>\n",
       "      <th>Array.129</th>\n",
       "      <th>Array.34</th>\n",
       "      <th>Array.67</th>\n",
       "      <th>Array.24</th>\n",
       "      <th>Array.22</th>\n",
       "      <th>Array.36</th>\n",
       "      <th>...</th>\n",
       "      <th>Array.64</th>\n",
       "      <th>Array.89</th>\n",
       "      <th>Array.30</th>\n",
       "      <th>Array.35</th>\n",
       "      <th>Array.93</th>\n",
       "      <th>Array.10</th>\n",
       "      <th>Array.123</th>\n",
       "      <th>Array.100</th>\n",
       "      <th>Array.134</th>\n",
       "      <th>Array.130</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>HER2+</td>\n",
       "      <td>HR+</td>\n",
       "      <td>HR+</td>\n",
       "      <td>Triple Neg</td>\n",
       "      <td>Triple Neg</td>\n",
       "      <td>HR+</td>\n",
       "      <td>...</td>\n",
       "      <td>Triple Neg</td>\n",
       "      <td>HR+</td>\n",
       "      <td>Triple Neg</td>\n",
       "      <td>HR+</td>\n",
       "      <td>Triple Neg</td>\n",
       "      <td>HER2+</td>\n",
       "      <td>HR+</td>\n",
       "      <td>HR+</td>\n",
       "      <td>HR+</td>\n",
       "      <td>HER2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>85022</td>\n",
       "      <td>216735</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>370546</td>\n",
       "      <td>372295</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>471671</td>\n",
       "      <td>786483</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>792533</td>\n",
       "      <td>907406</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 104 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome   Start     End Nclone Array.129 Array.34 Array.67    Array.24  \\\n",
       "0                                       HER2+      HR+      HR+  Triple Neg   \n",
       "1          1   85022  216735      4         0        0        0           0   \n",
       "2          1  370546  372295      4         0        0        0           0   \n",
       "3          1  471671  786483      5         0        0        0           0   \n",
       "4          1  792533  907406     13         0        0        0           0   \n",
       "\n",
       "     Array.22 Array.36    ...       Array.64 Array.89    Array.30 Array.35  \\\n",
       "0  Triple Neg      HR+    ...     Triple Neg      HR+  Triple Neg      HR+   \n",
       "1           0        0    ...              0        0           1        0   \n",
       "2           0        0    ...              0        0           1        0   \n",
       "3           0        0    ...              0        1           1        0   \n",
       "4           0        0    ...              0        1           1        0   \n",
       "\n",
       "     Array.93 Array.10 Array.123 Array.100 Array.134 Array.130  \n",
       "0  Triple Neg    HER2+       HR+       HR+       HR+     HER2+  \n",
       "1           0        0         0         0        -1         0  \n",
       "2           1        0         0         0        -1         0  \n",
       "3           1        0         0         0        -1         0  \n",
       "4           1        0         0         0        -1         0  \n",
       "\n",
       "[5 rows x 104 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgroup = ['', '', '', '']\n",
    "for n in range(4, 104):    \n",
    "    if df2.Sample[n-4] == df1.columns[n]:    \n",
    "        cgroup.append(df2.Subgroup[n-4])\n",
    "df = df1\n",
    "df.loc[0] = cgroup\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Subgroup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Array.129</td>\n",
       "      <td>HER2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Array.34</td>\n",
       "      <td>HR+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Array.67</td>\n",
       "      <td>HR+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Array.24</td>\n",
       "      <td>Triple Neg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Array.22</td>\n",
       "      <td>Triple Neg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample    Subgroup\n",
       "0  Array.129       HER2+\n",
       "1   Array.34         HR+\n",
       "2   Array.67         HR+\n",
       "3   Array.24  Triple Neg\n",
       "4   Array.22  Triple Neg"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#explore df2\n",
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#transpose the df1\n",
    "df1_transposed = df1.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>2824</th>\n",
       "      <th>2825</th>\n",
       "      <th>2826</th>\n",
       "      <th>2827</th>\n",
       "      <th>2828</th>\n",
       "      <th>2829</th>\n",
       "      <th>2830</th>\n",
       "      <th>2831</th>\n",
       "      <th>2832</th>\n",
       "      <th>2833</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Chromosome</th>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td></td>\n",
       "      <td>85022</td>\n",
       "      <td>370546</td>\n",
       "      <td>471671</td>\n",
       "      <td>792533</td>\n",
       "      <td>912799</td>\n",
       "      <td>1271190</td>\n",
       "      <td>1676445</td>\n",
       "      <td>1738295</td>\n",
       "      <td>2481927</td>\n",
       "      <td>...</td>\n",
       "      <td>151067607</td>\n",
       "      <td>152422390</td>\n",
       "      <td>152552851</td>\n",
       "      <td>152576854</td>\n",
       "      <td>152994680</td>\n",
       "      <td>153062077</td>\n",
       "      <td>153466463</td>\n",
       "      <td>153504394</td>\n",
       "      <td>153938998</td>\n",
       "      <td>153997146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td></td>\n",
       "      <td>216735</td>\n",
       "      <td>372295</td>\n",
       "      <td>786483</td>\n",
       "      <td>907406</td>\n",
       "      <td>1266212</td>\n",
       "      <td>1590570</td>\n",
       "      <td>1703748</td>\n",
       "      <td>2477597</td>\n",
       "      <td>2562342</td>\n",
       "      <td>...</td>\n",
       "      <td>152416606</td>\n",
       "      <td>152548587</td>\n",
       "      <td>152570071</td>\n",
       "      <td>152935130</td>\n",
       "      <td>153054487</td>\n",
       "      <td>153452633</td>\n",
       "      <td>153491568</td>\n",
       "      <td>153933426</td>\n",
       "      <td>153989329</td>\n",
       "      <td>154492924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nclone</th>\n",
       "      <td></td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>96</td>\n",
       "      <td>45</td>\n",
       "      <td>4</td>\n",
       "      <td>104</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>159</td>\n",
       "      <td>20</td>\n",
       "      <td>6</td>\n",
       "      <td>64</td>\n",
       "      <td>5</td>\n",
       "      <td>57</td>\n",
       "      <td>4</td>\n",
       "      <td>55</td>\n",
       "      <td>5</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Array.129</th>\n",
       "      <td>HER2+</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 2834 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             0       1       2       3       4        5        6        7     \\\n",
       "Chromosome              1       1       1       1        1        1        1   \n",
       "Start               85022  370546  471671  792533   912799  1271190  1676445   \n",
       "End                216735  372295  786483  907406  1266212  1590570  1703748   \n",
       "Nclone                  4       4       5      13       96       45        4   \n",
       "Array.129   HER2+       0       0       0       0        0        0        0   \n",
       "\n",
       "               8        9       ...           2824       2825       2826  \\\n",
       "Chromosome        1        1    ...             23         23         23   \n",
       "Start       1738295  2481927    ...      151067607  152422390  152552851   \n",
       "End         2477597  2562342    ...      152416606  152548587  152570071   \n",
       "Nclone          104       14    ...            159         20          6   \n",
       "Array.129         0        0    ...              2          2          2   \n",
       "\n",
       "                 2827       2828       2829       2830       2831       2832  \\\n",
       "Chromosome         23         23         23         23         23         23   \n",
       "Start       152576854  152994680  153062077  153466463  153504394  153938998   \n",
       "End         152935130  153054487  153452633  153491568  153933426  153989329   \n",
       "Nclone             64          5         57          4         55          5   \n",
       "Array.129           2          0          1          1          1          1   \n",
       "\n",
       "                 2833  \n",
       "Chromosome         23  \n",
       "Start       153997146  \n",
       "End         154492924  \n",
       "Nclone             43  \n",
       "Array.129           1  \n",
       "\n",
       "[5 rows x 2834 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1_transposed.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df1\n",
    "#### For df1, chromosome, start, end and #clone are self explanatory. The number of arrays varies and 100 are incuded. Arrays value is [-1, 0, 1, 2]. (-1: loss, 0: normal, 1: gain, 2: amplification (++gain)).\n",
    "### df2\n",
    "#### df2 assigns a certain array to a subgroup of breast cancer. The subtypes are three: HER2+, HR+, TN.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#check the values for each array\n",
    "def value_of_arrays(x):\n",
    "    '''\n",
    "    x: number of array\n",
    "    output: name of the array, the numbers for loss, normal, gain, amplification\n",
    "    and the group of breast cancer\n",
    "    '''\n",
    "    loss = norm = gain = dgain = 0\n",
    "    values_array = [loss, norm, gain, dgain]\n",
    "    \n",
    "    #count all the array values\n",
    "    for index, row in df1.iterrows():\n",
    "        if row[x]== -1:\n",
    "             values_array[0] += 1\n",
    "        elif row[x]== 0:\n",
    "            values_array[1] += 1\n",
    "        elif row[x]== 1:\n",
    "            values_array[2] += 1\n",
    "        elif row[x]== 2:\n",
    "            values_array[3] += 1\n",
    "            \n",
    "    #create the group of cancer type       \n",
    "    for index, row in df2.iterrows():\n",
    "        if row['Sample'] == df1.columns.values[x]:\n",
    "            group = row['Subgroup']\n",
    "        else:\n",
    "            pass\n",
    "    \n",
    "    return df1.columns.values[x], values_array, group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function for the amount of array values per chromosome\n",
    "def array_values_per_chrom(chromosome):\n",
    "\n",
    "    normal = loss = gain = dgain = count = 0\n",
    "    for j in df1.columns.values:\n",
    "        if j == \"Chromosome\" or j == \"Start\" or j == \"End\" or j == \"Nclone\":\n",
    "            continue\n",
    "        else:\n",
    "            for i in range(df1.shape[0]):\n",
    "                if df1.Chromosome[i] == chromosome:\n",
    "                    if df1[j][i] == 0:\n",
    "                        normal += 1\n",
    "                    elif df1[j][i] == -1:\n",
    "                        loss += 1\n",
    "                    elif df1[j][i] == 1:\n",
    "                        gain += 1\n",
    "                    elif df1[j][i] == 2:\n",
    "                        dgain += 1\n",
    "                \n",
    "                if df1.Chromosome[i] == chromosome:\n",
    "                    count+=1\n",
    "                        \n",
    "    return loss, normal, gain, dgain, int(count/100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 (4531, 14057, 8187, 325, 271)\n",
      "chr2 (980, 7714, 1401, 5, 101)\n",
      "chr3 (2008, 7804, 1958, 30, 118)\n",
      "chr4 (2454, 8891, 1106, 49, 125)\n",
      "chr5 (1914, 6205, 1816, 65, 100)\n",
      "chr6 (4744, 9900, 3304, 152, 181)\n",
      "chr7 (749, 6735, 2637, 79, 102)\n",
      "chr8 (5136, 7882, 4569, 1713, 193)\n",
      "chr9 (2367, 6038, 1226, 69, 97)\n",
      "chr10 (2415, 7354, 1759, 72, 116)\n",
      "chr11 (3472, 12146, 2936, 246, 188)\n",
      "chr12 (1183, 7936, 2299, 82, 115)\n",
      "chr13 (1863, 3894, 636, 7, 64)\n",
      "chr14 (2408, 6469, 1581, 142, 106)\n",
      "chr15 (1713, 4520, 823, 44, 71)\n",
      "chr16 (1847, 5331, 2069, 53, 93)\n",
      "chr17 (8919, 18399, 5965, 1117, 344)\n",
      "chr18 (1620, 3175, 795, 10, 56)\n",
      "chr19 (1427, 7284, 1629, 60, 104)\n",
      "chr20 (1114, 6258, 3452, 276, 111)\n",
      "chr21 (431, 2872, 994, 3, 43)\n",
      "chr22 (1689, 3634, 772, 5, 61)\n",
      "chr23 (161, 984, 5357, 898, 74)\n"
     ]
    }
   ],
   "source": [
    "for i in range(1, 24):\n",
    "    print('chr'+str(i),array_values_per_chrom(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function for exploring the type of cancer correlated with their array values\n",
    "def bcancer_type_arrays(ctype, amount = False, arrays = False):\n",
    "    '''\n",
    "    ctype: \"Triple Neg\", \"HR+\", \"HER2+\"\n",
    "    '''\n",
    "    loss = norm = gain = dgain = count = 0\n",
    "    for i in range(4,df1.shape[1]):\n",
    "        if value_of_arrays(i)[2] == ctype:\n",
    "            if arrays == True:\n",
    "                print(value_of_arrays(i))\n",
    "            count +=1\n",
    "            loss +=value_of_arrays(i)[1][0]\n",
    "            norm +=value_of_arrays(i)[1][1]\n",
    "            gain +=value_of_arrays(i)[1][2]\n",
    "            dgain +=value_of_arrays(i)[1][3]\n",
    "        else:\n",
    "            pass\n",
    "        #if value_of_arrays(i)\n",
    "        \n",
    "    if amount == True:\n",
    "        print(count)\n",
    "    return int(loss/count), int(norm/count), int(gain/count), int(dgain/count)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(592, 1530, 655, 55)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bcancer_type_arrays(\"Triple Neg\",True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(578, 1653, 538, 62)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bcancer_type_arrays(\"HER2+\",True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(490, 1766, 529, 47)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bcancer_type_arrays(\"HR+\",True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Based on the above you cannot reach a conclusion only by looking the array values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find correlation between regions of DNA and values of array\n",
    "def regions_DNA_values(x):\n",
    "    '''\n",
    "    x: number of column\n",
    "    output: Chromosome, Start, End, Nclones, and array values\n",
    "    '''\n",
    "    loss = norm = gain = dgain = 0\n",
    "    values_region = [loss, norm, gain, dgain]\n",
    "\n",
    "    for index, row in df1_transposed.iterrows():\n",
    "        if index == \"Chromosome\" or index == \"Start\" or index == \"End\" or index == \"Nclone\":\n",
    "            continue\n",
    "        else: \n",
    "            if row[x]== -1:\n",
    "                values_region[0] += 1\n",
    "            elif row[x]== 0:\n",
    "                values_region[1] += 1\n",
    "            elif row[x]== 1:\n",
    "                values_region[2] += 1\n",
    "            elif row[x]== 2:\n",
    "                values_region[3] += 1\n",
    "    return df1_transposed[x][0], df1_transposed[x][1], df1_transposed[x][2], df1_transposed[x][3], values_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1, 2903052, 3112977, 27, [25, 67, 8, 0])\n"
     ]
    }
   ],
   "source": [
    "#for i in range(df1_transposed.shape[1]):\n",
    "    print(regions_DNA_values(18))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[494, 269, 310, 501, 496, 451, 313]\n"
     ]
    }
   ],
   "source": [
    "#count the number of clones\n",
    "#maybe the amount of clones means more reliability\n",
    "ufive = uten = utwenty = ufifty = uhundred = u2hundred = more = 0\n",
    "Nclone = [ufive, uten, utwenty, ufifty, uhundred, u2hundred, more]\n",
    "for i in range(0,2834):\n",
    "    if regions_DNA_values(i)[3] <= 5:\n",
    "        Nclone[0] +=1\n",
    "    elif regions_DNA_values(i)[3] <= 10:\n",
    "        Nclone[1] +=1\n",
    "    elif regions_DNA_values(i)[3] <= 20:\n",
    "        Nclone[2] +=1\n",
    "    elif regions_DNA_values(i)[3] <= 50:\n",
    "        Nclone[3] +=1\n",
    "    elif regions_DNA_values(i)[3] <= 100:\n",
    "        Nclone[4] +=1\n",
    "    elif regions_DNA_values(i)[3] <= 200:\n",
    "        Nclone[5] +=1\n",
    "    elif regions_DNA_values(i)[3] > 200:\n",
    "        Nclone[6] +=1\n",
    "    else:\n",
    "        pass\n",
    "print(Nclone)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def chromosome_array_values(array_value, threshold, plot = False):\n",
    "    '''\n",
    "    array_value: 0: -1 = loss, 1: 0 = normal, 2: 1 = gain, 3: 2 = 2gain\n",
    "    threshold: number out of a hundred\n",
    "    plot: True/False if you need a bar plot\n",
    "    '''\n",
    "    dic = {}\n",
    "    one = two = three = four = five = six = seven = eight = nine = ten = eleven = twelve = thirteen = fourteen = 0\n",
    "    fifteen = sixteen = seventeen = eighteen = nineteen = twenty = twentyone = twentytwo = twentythree = 0\n",
    "    chrom = [one, two, three, four, five, six, seven, eight, nine, ten, eleven, twelve, thirteen, fourteen, \n",
    "           fifteen, sixteen, seventeen, eighteen, nineteen, twenty, twentyone, twentytwo, twentythree]\n",
    "    keys = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17',\n",
    "            '18', '19', '20', '21', '22', '23']\n",
    "    \n",
    "    #iterate throught all the given DNA regions \n",
    "    for i in range(0,df1_transposed.shape[1]):\n",
    "        if regions_DNA_values(i)[4][array_value] > threshold:\n",
    "            #assign the array values to each chromosome\n",
    "            for chm in range(1,len(chrom)): \n",
    "                if regions_DNA_values(i)[0] == chm:\n",
    "                    chrom[chm-1] += 1\n",
    "                else:\n",
    "                    pass\n",
    "            if regions_DNA_values(i)[0] == 23: #hardcoded this part, caused I cannot fixed it else now\n",
    "                    chrom[22] += 1\n",
    "            print(regions_DNA_values(i))\n",
    "        else:\n",
    "            pass\n",
    "\n",
    "    #create a dictionary to store the amount of array values per chromosome(key)\n",
    "    for n in range(len(chrom)):\n",
    "        dic[keys[n]] = chrom[n]\n",
    "    \n",
    "    #bar plot creation \n",
    "    if plot == True:\n",
    "        plt.bar(range(len(dic)), list(dic.values()), align='center')\n",
    "        plt.xticks(range(len(dic)), list(dic.keys()))\n",
    "        plt.show()\n",
    "    else:\n",
    "        pass\n",
    "\n",
    "    return dic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3, 163997236, 164023510, 2, [56, 27, 16, 1])\n",
      "(3, 164038925, 164101784, 6, [71, 18, 10, 1])\n",
      "(4, 69215734, 69311985, 10, [64, 26, 10, 0])\n",
      "(6, 32586131, 32600962, 3, [79, 10, 11, 0])\n",
      "(6, 32605300, 32629848, 2, [73, 16, 11, 0])\n",
      "(7, 141965869, 141978333, 4, [57, 28, 15, 0])\n",
      "(8, 166452, 181530, 3, [59, 37, 4, 0])\n",
      "(8, 184617, 808139, 63, [63, 33, 4, 0])\n",
      "(8, 822927, 1136619, 36, [63, 32, 5, 0])\n",
      "(8, 1152551, 2563316, 132, [62, 34, 4, 0])\n",
      "(8, 2586264, 3627077, 113, [63, 34, 3, 0])\n",
      "(8, 3635646, 4294791, 74, [64, 33, 3, 0])\n",
      "(8, 4299996, 4692082, 48, [65, 31, 4, 0])\n",
      "(8, 4708444, 5657660, 43, [64, 33, 3, 0])\n",
      "(8, 5677578, 5923347, 12, [64, 32, 4, 0])\n",
      "(8, 5952862, 6907624, 98, [64, 31, 5, 0])\n",
      "(8, 8148437, 11897919, 356, [65, 30, 5, 0])\n",
      "(8, 11903901, 12448374, 5, [60, 35, 5, 0])\n",
      "(8, 12511855, 12602139, 4, [59, 36, 5, 0])\n",
      "(8, 12623479, 14185328, 128, [66, 30, 4, 0])\n",
      "(8, 14193778, 14291793, 13, [66, 31, 3, 0])\n",
      "(8, 14298266, 16315130, 119, [65, 31, 3, 1])\n",
      "(8, 16373516, 16668380, 15, [66, 31, 2, 1])\n",
      "(8, 16701627, 17310921, 58, [64, 31, 4, 1])\n",
      "(8, 17316883, 18940172, 157, [65, 31, 4, 0])\n",
      "(8, 18983405, 19153325, 9, [65, 31, 3, 1])\n",
      "(8, 19181564, 21447923, 139, [65, 32, 3, 0])\n",
      "(8, 21490551, 22657603, 149, [66, 31, 3, 0])\n",
      "(8, 22663858, 23132039, 58, [67, 31, 2, 0])\n",
      "(8, 23137151, 25096564, 149, [65, 34, 1, 0])\n",
      "(8, 25114132, 25774629, 65, [64, 34, 2, 0])\n",
      "(8, 25782621, 26587838, 77, [64, 34, 2, 0])\n",
      "(8, 26613950, 27431467, 73, [62, 35, 3, 0])\n",
      "(8, 27439677, 28182114, 81, [60, 36, 3, 1])\n",
      "(8, 28208003, 28538941, 33, [63, 33, 4, 0])\n",
      "(8, 28568337, 28877291, 32, [61, 33, 5, 1])\n",
      "(8, 28883487, 29205723, 37, [60, 33, 6, 1])\n",
      "(8, 29233976, 29264810, 6, [58, 36, 5, 1])\n",
      "(8, 29281419, 29430834, 5, [56, 36, 7, 1])\n",
      "(8, 29497110, 29571339, 6, [57, 35, 7, 1])\n",
      "(8, 29592601, 30101874, 30, [59, 34, 7, 0])\n",
      "(8, 30109992, 30278937, 14, [58, 35, 7, 0])\n",
      "(8, 30302476, 30776804, 55, [57, 38, 5, 0])\n",
      "(8, 30786432, 30939857, 12, [57, 38, 4, 1])\n",
      "(8, 30952516, 31295499, 36, [58, 38, 4, 0])\n",
      "(8, 31336091, 31896351, 42, [58, 36, 5, 1])\n",
      "(8, 31906730, 32154763, 28, [58, 37, 5, 0])\n",
      "(8, 32161837, 32407642, 32, [58, 36, 5, 1])\n",
      "(8, 32418860, 32553445, 16, [55, 38, 6, 1])\n",
      "(8, 32567438, 32617200, 8, [56, 38, 5, 1])\n",
      "(8, 32625061, 32786232, 18, [56, 38, 5, 1])\n",
      "(8, 32812909, 32947324, 7, [53, 40, 6, 1])\n",
      "(8, 32962484, 33337952, 16, [53, 39, 7, 1])\n",
      "(8, 33344787, 34072415, 62, [55, 38, 6, 1])\n",
      "(8, 34104577, 34785140, 34, [57, 36, 6, 1])\n",
      "(8, 34801779, 34987140, 8, [55, 35, 9, 1])\n",
      "(8, 35005401, 35166837, 9, [52, 37, 10, 1])\n",
      "(11, 55124730, 55194990, 11, [56, 36, 7, 1])\n",
      "(11, 70519863, 70552702, 6, [54, 31, 13, 2])\n",
      "(14, 19402284, 19490630, 15, [55, 32, 12, 1])\n",
      "(14, 73071404, 73092018, 4, [61, 30, 9, 0])\n",
      "(14, 105857193, 105881279, 4, [53, 38, 8, 1])\n",
      "(16, 68710277, 68751390, 5, [55, 35, 9, 1])\n",
      "(17, 48539, 59739, 2, [51, 45, 4, 0])\n",
      "(17, 66107, 390571, 43, [54, 43, 3, 0])\n",
      "(17, 465054, 974692, 73, [51, 46, 3, 0])\n",
      "(17, 1008443, 1295205, 34, [52, 45, 3, 0])\n",
      "(17, 1302415, 1427745, 18, [51, 46, 3, 0])\n",
      "(17, 1432975, 1439166, 3, [52, 46, 2, 0])\n",
      "(17, 1440363, 1452762, 4, [51, 46, 3, 0])\n",
      "(17, 1456417, 1524691, 12, [51, 45, 4, 0])\n",
      "(17, 1562322, 1703117, 23, [51, 45, 4, 0])\n",
      "(17, 1912424, 2133315, 24, [51, 45, 4, 0])\n",
      "(17, 2145322, 2286311, 23, [51, 44, 5, 0])\n",
      "(17, 2333732, 2343255, 4, [57, 42, 1, 0])\n",
      "(17, 2361350, 2900841, 61, [53, 43, 4, 0])\n",
      "(17, 2914621, 3387328, 60, [54, 43, 3, 0])\n",
      "(17, 3393853, 4388997, 123, [52, 45, 3, 0])\n",
      "(17, 4393276, 4966472, 93, [54, 43, 3, 0])\n",
      "(17, 4974667, 5014217, 5, [51, 45, 4, 0])\n",
      "(17, 5031837, 5332851, 41, [55, 42, 3, 0])\n",
      "(17, 5343504, 7517491, 247, [57, 40, 3, 0])\n",
      "(17, 7522884, 8559106, 145, [57, 40, 3, 0])\n",
      "(17, 8567837, 9872633, 143, [57, 39, 4, 0])\n",
      "(17, 9885625, 9934563, 7, [56, 39, 4, 1])\n",
      "(17, 9943837, 11456680, 128, [57, 39, 4, 0])\n",
      "(17, 11469879, 12854787, 119, [57, 40, 3, 0])\n",
      "(17, 12861144, 13000886, 6, [55, 40, 5, 0])\n",
      "(17, 13035306, 15423538, 134, [56, 41, 3, 0])\n",
      "(17, 15462570, 16521872, 101, [56, 41, 3, 0])\n",
      "(17, 16578627, 16713678, 5, [52, 44, 3, 1])\n",
      "(17, 16733547, 17098966, 42, [53, 44, 3, 0])\n",
      "(17, 17104943, 17885418, 81, [55, 42, 3, 0])\n",
      "(17, 17898157, 18224296, 43, [53, 44, 3, 0])\n",
      "(17, 18474499, 18868037, 32, [54, 41, 5, 0])\n",
      "(17, 19083766, 19526967, 49, [52, 41, 7, 0])\n",
      "(17, 19542146, 19715593, 22, [51, 42, 6, 1])\n",
      "(17, 20234571, 20521986, 8, [51, 44, 5, 0])\n",
      "(17, 41527705, 41566540, 6, [64, 27, 8, 1])\n",
      "(17, 41577520, 41644979, 7, [77, 17, 6, 0])\n",
      "(17, 41700762, 41706870, 2, [75, 19, 6, 0])\n",
      "(22, 22681244, 22690716, 2, [59, 27, 14, 0])\n",
      "(22, 22695759, 22714749, 5, [63, 23, 14, 0])\n",
      "(22, 37683612, 37705310, 5, [60, 34, 6, 0])\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4xLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvAOZPmwAAEUhJREFUeJzt3XuspHV9x/H3110oFy+AnKUrCz3UrARiKuApwaJYF7UIZsFWjMaabcRsatSC1upSGqOxTRav7R+NhgJ14xWKIIhW2a6gbaLoWa6Li6K44sq6e7xQtSYq8u0fz2/bk+OZ8/xmzmWXH+9XMpnnmfP77u/LzG8+85xnZg6RmUiSHv0et68bkCQtDANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1IjlSznZkUcemePj40s5pSQ96m3duvWHmTnWN25JA318fJzJycmlnFKSHvUi4rs14zzlIkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5Jjaj6pmhE7AB+BvwGeDgzJyLiCOAqYBzYAbwsM3+yOG1qXxjf8JnqsTs2nrOInUiqMcwR+vMy86TMnCj7G4Atmbka2FL2JUn7yHxOuZwLbCrbm4Dz5t+OJGlUtYGewE0RsTUi1pfbjsrMXQDlesViNChJqlP71xZPz8wHI2IFsDki7q2doLwArAc49thjR2hRklSj6gg9Mx8s13uA64BTgd0RsRKgXO8ZUHtZZk5k5sTYWO+f85Ukjag30CPi0Ih4wt5t4IXANuAGYF0Ztg64frGalCT1qznlchRwXUTsHf+xzPxcRHwNuDoiLgAeAM5fvDYlSX16Az0z7weeMcvtPwLOXIymJEnD85uiktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEZUB3pELIuI2yPixrJ/XETcGhH3RcRVEXHg4rUpSeozzBH6hcD2afuXAu/PzNXAT4ALFrIxSdJwqgI9IlYB5wCXl/0A1gDXlCGbgPMWo0FJUp3aI/R/BN4CPFL2nww8lJkPl/2dwNGzFUbE+oiYjIjJqampeTUrSRqsN9Aj4sXAnszcOv3mWYbmbPWZeVlmTmTmxNjY2IhtSpL6LK8YczqwNiLOBg4Cnkh3xH5YRCwvR+mrgAcXr01JUp/eI/TMvDgzV2XmOPBy4AuZ+UrgZuClZdg64PpF61KS1Gs+n0N/K/CmiPgW3Tn1KxamJUnSKGpOufyfzLwFuKVs3w+cuvAtSZJG4TdFJakRBrokNcJAl6RGDHUOXVJbxjd8pmrcjo3nLHInWggeoUtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjegN9Ig4KCK+GhF3RsQ9EfGOcvtxEXFrRNwXEVdFxIGL364kaZCaI/RfAmsy8xnAScBZEXEacCnw/sxcDfwEuGDx2pQk9ekN9Oz8vOweUC4JrAGuKbdvAs5blA4lSVWqzqFHxLKIuAPYA2wGvg08lJkPlyE7gaMXp0VJUo2qQM/M32TmScAq4FTghNmGzVYbEesjYjIiJqempkbvVJI0p6E+5ZKZDwG3AKcBh0XE8vKjVcCDA2ouy8yJzJwYGxubT6+SpDnUfMplLCIOK9sHA88HtgM3Ay8tw9YB1y9Wk5Kkfsv7h7AS2BQRy+heAK7OzBsj4uvAJyLi74HbgSsWsU9JUo/eQM/Mu4CTZ7n9frrz6ZKk/YDfFJWkRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1ojfQI+KYiLg5IrZHxD0RcWG5/YiI2BwR95Xrwxe/XUnSIDVH6A8Df52ZJwCnAa+LiBOBDcCWzFwNbCn7kqR9pDfQM3NXZt5Wtn8GbAeOBs4FNpVhm4DzFqtJSVK/oc6hR8Q4cDJwK3BUZu6CLvSBFQvdnCSpXnWgR8TjgU8CF2XmT4eoWx8RkxExOTU1NUqPkqQKVYEeEQfQhflHM/PacvPuiFhZfr4S2DNbbWZelpkTmTkxNja2ED1LkmZR8ymXAK4Atmfm+6b96AZgXdleB1y/8O1JkmotrxhzOvAq4O6IuKPc9rfARuDqiLgAeAA4f3FalCTV6A30zPwvIAb8+MyFbUeSNCq/KSpJjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqRG+gR8SVEbEnIrZNu+2IiNgcEfeV68MXt01JUp+aI/QPAWfNuG0DsCUzVwNbyr4kaR/qDfTM/BLw4xk3nwtsKtubgPMWuC9J0pBGPYd+VGbuAijXKwYNjIj1ETEZEZNTU1MjTidJ6rPob4pm5mWZOZGZE2NjY4s9nSQ9Zo0a6LsjYiVAud6zcC1JkkYxaqDfAKwr2+uA6xemHUnSqGo+tvhx4MvA8RGxMyIuADYCL4iI+4AXlH1J0j60vG9AZr5iwI/OXOBeJEnz4DdFJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWrE8n3dgNSy8Q2fqRq3Y+M5i9yJ5lL7OMH+/Vh5hC5JjTDQJakRBrokNcJz6LPY38977u/9LSXvC+n/eYQuSY0w0CWpEQa6JDViXufQI+Is4J+AZcDlmblxQbp6FNrfP8e6v/en+fHxFczjCD0ilgH/DLwIOBF4RUScuFCNSZKGM59TLqcC38rM+zPzV8AngHMXpi1J0rDmE+hHA9+btr+z3CZJ2gciM0crjDgf+JPMfE3ZfxVwama+Yca49cD6sns88I3R2/0tRwI/XKK6papZyrn29/6Wci77W/qapZxrf++vz+9l5ljvqMwc6QI8C/j8tP2LgYtH/fdG7GFyqeqWqsb+vC/sb/+oWeq5FuIyn1MuXwNWR8RxEXEg8HLghnn8e5KkeRj5Y4uZ+XBEvB74PN3HFq/MzHsWrDNJ0lDm9Tn0zPws8NkF6mUUly1h3VLVLOVc+3t/SzmX/S19zVLOtb/3tyBGflNUkrR/8av/ktSIR2WgR8SVEbEnIrYNUXNMRNwcEdsj4p6IuLCy7qCI+GpE3Fnq3jHEnMsi4vaIuLFy/I6IuDsi7oiIySHmOSwiromIe8t/37N6xh9f5th7+WlEXFQxzxvLfbAtIj4eEQdV9ndhqbln0DyzPaYRcUREbI6I+8r14ZV155e5HomIicqad5f7766IuC4iDquoeWcZf0dE3BQRT6mZa9rP3hwRGRFHVsz19oj4/rTH7OyaeSLiDRHxjXJ/vKvyvrhq2jw7IuKOipqTIuIre9duRJxaUfOMiPhyWfOfjognzqiZ9Tnbty7mqBu4Luao6VsXg+p618ai2Fcfr5nPBTgDOAXYNkTNSuCUsv0E4JvAiRV1ATy+bB8A3AqcVjnnm4CPATdWjt8BHDnC/bEJeE3ZPhA4bIjaZcAP6D7nOte4o4HvAAeX/auBv6j4958ObAMOoXvP5j+A1TWPKfAuYEPZ3gBcWll3At13Hm4BJiprXggsL9uXzpxrQM0Tp23/FfDB2rUKHEP3gYLvznzMB8z1duDNwzwngOeV+/t3yv6K2v6m/fy9wNsq5roJeFHZPhu4paLma8Bzy/argXfOqJn1Odu3LuaoG7gu5qjpWxeD6nrXxmJcHpVH6Jn5JeDHQ9bsyszbyvbPgO1UfLM1Oz8vuweUS+8bDxGxCjgHuHyYPodVjmrOAK4AyMxfZeZDQ/wTZwLfzszvVoxdDhwcEcvpAvrBipoTgK9k5i8y82Hgi8BLZg4a8JieS/diRbk+r6YuM7dn5sAvsA2ouan0B/AVYFVFzU+n7R7KLOtijrX6fuAtQ9YMNKDmtcDGzPxlGbNnmLkiIoCXAR+vqElg7xH2k5ixNgbUHA98qWxvBv5sRs2g5+yc62JQ3VzrYo6avnUxqK53bSyGR2Wgz1dEjAMn0x1t14xfVn7t3ANszsyaun+ke8I+MkRrCdwUEVuj+4Ztjd8HpoB/je70zuURcegQc76cGU/YWRvL/D7wHuABYBfw35l5U8W/vw04IyKeHBGH0B29HVPZ21GZuavMvwtYUVk3X68G/r1mYET8Q0R8D3gl8LbKmrXA9zPzziH7en35Nf7K2U4/zeJpwHMi4taI+GJE/OGQ8z0H2J2Z91WMvQh4d7kv3kP3RcM+24C1Zft85lgXM56z1eti2Od6T82c62Jm3ShrY74ec4EeEY8HPglcNONVdKDM/E1mnkT36nxqRDy9Z44XA3syc+uQ7Z2emafQ/QXL10XEGRU1y+l+lf1AZp4M/A/dr6G9ovtC2Frg3yrGHk53ZHQc8BTg0Ij48766zNxO96vqZuBzwJ3Aw3MW7UMRcQldfx+tGZ+Zl2TmMWX86yv+/UOASxj+Cf4B4KnASXQvqO+tqFkOHA6cBvwNcHU56q71Cipe7IvXAm8s98UbKb8x9ng13TrfSne64lezDRrlOTtq3aCavnUxW92wa2MhPKYCPSIOoLvTP5qZ1w5bX05l3AKc1TP0dGBtROyg+yuUayLiIxX//oPleg9wHd1ftOyzE9g57beGa+gCvsaLgNsyc3fF2OcD38nMqcz8NXAt8Ec1k2TmFZl5SmaeQfdrd80RH8DuiFgJUK5/65TBQoqIdcCLgVdmOfk5hI8x45TBAE+le1G8s6yPVcBtEfG7cxVl5u5yYPEI8C/Ur41ry2nDr9L9tnhkTw0A5bTanwJX1YwH1tGtCegOEHr7y8x7M/OFmflMuheOb8/Sx2zP2d51McpzfVBN37qomKt2bczbYybQy5HJFcD2zHzfEHVje9/ZjoiD6YLt3rlqMvPizFyVmeN0pzS+kJlzHs1GxKER8YS923RvxvR+iiczfwB8LyKOLzedCXy9r64Y5gjsAeC0iDik3Jdn0p0v7BURK8r1sXQhUTvnDXRBQbm+vrJuaNH9z1reCqzNzF9U1qyetruWnnUBkJl3Z+aKzBwv62Mn3ZtqP+iZa+W03ZdQsTaATwFrSv3T6N4wr/2jUc8H7s3MnZXjHwSeW7bXUPGiPW1dPA74O+CDM34+6Dk757oY5bk+qKZvXcxRN/TaWBCD3i3dny90gbAL+DXdE+KCippn052jvgu4o1zOrqj7A+D2UreNGe/4V9T/MRWfcqE7F35nudwDXDLEHCcBk6XHTwGHV9QcAvwIeNIQ87yDbmFuAz5M+fRERd1/0r3I3AmcWfuYAk8GttCFwxbgiMq6l5TtXwK7mfZH5Oao+Rbdn4PeuzY+WFHzyXJf3AV8mu7NsKHWKrN8smnAXB8G7i5z3QCsrKg5EPhI6fE2YE1tf8CHgL8c4rF6NrC1PMa3As+sqLmQ7lMh3wQ2Ur7o2Pec7VsXc9QNXBdz1PSti0F1vWtjMS5+U1SSGvGYOeUiSa0z0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJasT/ApC8m7+UdOhpAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f7cb67ab2e8>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "{'1': 0,\n",
       " '2': 0,\n",
       " '3': 2,\n",
       " '4': 1,\n",
       " '5': 0,\n",
       " '6': 2,\n",
       " '7': 1,\n",
       " '8': 51,\n",
       " '9': 0,\n",
       " '10': 0,\n",
       " '11': 2,\n",
       " '12': 0,\n",
       " '13': 0,\n",
       " '14': 3,\n",
       " '15': 0,\n",
       " '16': 1,\n",
       " '17': 38,\n",
       " '18': 0,\n",
       " '19': 0,\n",
       " '20': 0,\n",
       " '21': 0,\n",
       " '22': 3,\n",
       " '23': 0}"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chromosome_array_values(0, 50, True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 16 51 30 1\n",
      "chr2 9 76 13 0\n",
      "chr3 16 66 16 0\n",
      "chr4 19 71 8 0\n",
      "chr5 19 62 18 0\n",
      "chr6 26 54 18 0\n",
      "chr7 7 66 25 0\n",
      "chr8 26 40 23 8\n",
      "chr9 24 62 12 0\n",
      "chr10 20 63 15 0\n",
      "chr11 18 64 15 1\n",
      "chr12 10 68 19 0\n",
      "chr13 29 60 9 0\n",
      "chr14 22 60 14 1\n",
      "chr15 24 63 11 0\n",
      "chr16 20 57 22 0\n",
      "chr17 25 53 17 3\n",
      "chr18 28 56 14 0\n",
      "chr19 13 70 15 0\n",
      "chr20 9 56 31 2\n",
      "chr21 10 66 23 0\n",
      "chr22 28 59 12 0\n",
      "chr23 2 12 72 12\n"
     ]
    }
   ],
   "source": [
    "#for each chromosome print the array values\n",
    "chrloss = chrnorm = chrgain = chrdgain = count = 0\n",
    "for i in range(df1_transposed.shape[1]):\n",
    "    if i == 0:\n",
    "        chrloss +=regions_DNA_values(i)[4][0]\n",
    "        chrnorm +=regions_DNA_values(i)[4][1]\n",
    "        chrgain +=regions_DNA_values(i)[4][2]\n",
    "        chrdgain +=regions_DNA_values(i)[4][3]\n",
    "        count +=1\n",
    "    elif i == df1_transposed.shape[1]-1:\n",
    "        print('chr'+str(regions_DNA_values(i)[0]), int(chrloss/count), int(chrnorm/count), int(chrgain/count), int(chrdgain/count))\n",
    "    elif regions_DNA_values(i)[0] == regions_DNA_values(i-1)[0]:\n",
    "        chrloss +=regions_DNA_values(i)[4][0]\n",
    "        chrnorm +=regions_DNA_values(i)[4][1]\n",
    "        chrgain +=regions_DNA_values(i)[4][2]\n",
    "        chrdgain +=regions_DNA_values(i)[4][3]\n",
    "        count +=1\n",
    "    elif regions_DNA_values(i)[0] != regions_DNA_values(i-1)[0]:\n",
    "        print('chr'+str(regions_DNA_values(i-1)[0]), int(chrloss/count), int(chrnorm/count), int(chrgain/count), int(chrdgain/count))\n",
    "        chrloss = chrnorm = chrgain = chrdgain = count = 0\n",
    "    else:\n",
    "        print(\"unexpected\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Research questions:\n",
    "## 1) Assign a new array to a certain cancer type.\n",
    "## 2) Find out which regions are more responsible for certain cancer types than others.\n",
    "## 3) Find out which chromosomes are more responsible for certain cancer types than others (maybe too simple).\n",
    "## 4) Identify correlation between regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def create_list_for_cancer_types(ctype):\n",
    "    '''\n",
    "    ctype: \"HER2+\", \"HR+\", \"Triple Neg\"\n",
    "    '''\n",
    "    cancerlist = []     #it will contain lists for the cancer type you want\n",
    "    newt = []           #the list in which each array will be appended\n",
    "    count = amount_of_ctypes = 0        #for the array.number columns\n",
    "    for cancer in df1.values[0]:        #104 in total\n",
    "        if cancer == ctype:\n",
    "            amount_of_ctypes += 1\n",
    "            newt = [df1.columns[count]]\n",
    "            for regions in range(len(df1.values)):     #2834 total regions\n",
    "                newt.append(df.values[regions][count])\n",
    "            cancerlist.append(newt)\n",
    "        count+=1\n",
    "    return cancerlist, amount_of_ctypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def regions_with_values(threshold, ctype):\n",
    "    '''\n",
    "    threshold: between 0 and 1\n",
    "    ctype: \"HER2+\", \"HR+\", \"Triple Neg\"\n",
    "    '''\n",
    "    cancer_list = create_list_for_cancer_types(ctype)[0]\n",
    "    amount = create_list_for_cancer_types(ctype)[1]\n",
    "    #lossvalue = normalvalue = gainvalue = dgainvalue = 0\n",
    "    #losslist = normallist = gainlist = dgainlist = []\n",
    "    arraylist = []\n",
    "    for i in range(2, len(cancer_list[0])):\n",
    "        lossvalue = normalvalue = gainvalue = dgainvalue = 0\n",
    "        for j in range(len(cancer_list)):\n",
    "            if cancer_list[j][i] == -1:\n",
    "                lossvalue+=1\n",
    "            elif cancer_list[j][i] == 0:\n",
    "                normalvalue+=1\n",
    "            elif cancer_list[j][i] == 1:\n",
    "                gainvalue+=1\n",
    "            elif cancer_list[j][i] == 2:\n",
    "                dgainvalue+=1\n",
    "        #if i == (len(HER2list)-1):\n",
    "            if j == len(cancer_list)-1:\n",
    "                #print(i, lossvalue, normalvalue, gainvalue, dgainvalue)\n",
    "                if lossvalue > amount*threshold:\n",
    "                    region_val = [i, 'loss', lossvalue]\n",
    "                    arraylist.append(region_val)\n",
    "                elif normalvalue > amount*threshold:\n",
    "                    region_val = [i, 'normal', normalvalue]\n",
    "                    arraylist.append(region_val)\n",
    "                elif gainvalue > amount*threshold:\n",
    "                    region_val = [i, 'gain', gainvalue]\n",
    "                    arraylist.append(region_val)\n",
    "                elif dgainvalue > amount*threshold:\n",
    "                    region_val = [i, 'dgain', dgainvalue]                \n",
    "                    arraylist.append(region_val)\n",
    "                    #dgainlist.append(HER2list[j][0])\n",
    "    return arraylist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "hr_array = regions_with_values(0.5, \"HR+\")\n",
    "her2_array = regions_with_values(0.5, \"HER2+\")\n",
    "tn_array = regions_with_values(0.5, \"Triple Neg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def count_values(input_array):\n",
    "    return_list = []\n",
    "    for n in range(len(input_array)):\n",
    "        if input_array[n][1] == 'loss':\n",
    "            return_list.append(input_array[n])\n",
    "    return return_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[470, 'loss', 19],\n",
       " [471, 'loss', 26],\n",
       " [547, 'loss', 18],\n",
       " [672, 'loss', 17],\n",
       " [673, 'loss', 19],\n",
       " [674, 'loss', 18],\n",
       " [761, 'loss', 26],\n",
       " [762, 'loss', 24],\n",
       " [999, 'loss', 22],\n",
       " [1000, 'loss', 28],\n",
       " [1001, 'loss', 29],\n",
       " [1002, 'loss', 29],\n",
       " [1003, 'loss', 28],\n",
       " [1004, 'loss', 28],\n",
       " [1005, 'loss', 29],\n",
       " [1006, 'loss', 28],\n",
       " [1007, 'loss', 27],\n",
       " [1008, 'loss', 27],\n",
       " [1009, 'loss', 27],\n",
       " [1010, 'loss', 18],\n",
       " [1012, 'loss', 18],\n",
       " [1013, 'loss', 27],\n",
       " [1014, 'loss', 25],\n",
       " [1015, 'loss', 25],\n",
       " [1016, 'loss', 28],\n",
       " [1017, 'loss', 28],\n",
       " [1018, 'loss', 27],\n",
       " [1019, 'loss', 27],\n",
       " [1020, 'loss', 26],\n",
       " [1021, 'loss', 26],\n",
       " [1022, 'loss', 27],\n",
       " [1023, 'loss', 28],\n",
       " [1024, 'loss', 27],\n",
       " [1025, 'loss', 28],\n",
       " [1026, 'loss', 28],\n",
       " [1027, 'loss', 29],\n",
       " [1028, 'loss', 28],\n",
       " [1029, 'loss', 27],\n",
       " [1030, 'loss', 25],\n",
       " [1031, 'loss', 26],\n",
       " [1032, 'loss', 26],\n",
       " [1033, 'loss', 26],\n",
       " [1034, 'loss', 25],\n",
       " [1035, 'loss', 25],\n",
       " [1036, 'loss', 25],\n",
       " [1037, 'loss', 25],\n",
       " [1038, 'loss', 24],\n",
       " [1039, 'loss', 23],\n",
       " [1040, 'loss', 23],\n",
       " [1041, 'loss', 25],\n",
       " [1042, 'loss', 25],\n",
       " [1043, 'loss', 25],\n",
       " [1044, 'loss', 25],\n",
       " [1045, 'loss', 23],\n",
       " [1046, 'loss', 23],\n",
       " [1047, 'loss', 23],\n",
       " [1048, 'loss', 23],\n",
       " [1049, 'loss', 23],\n",
       " [1050, 'loss', 24],\n",
       " [1051, 'loss', 26],\n",
       " [1052, 'loss', 24],\n",
       " [1053, 'loss', 22],\n",
       " [1054, 'loss', 22],\n",
       " [1055, 'loss', 20],\n",
       " [1056, 'loss', 21],\n",
       " [1057, 'loss', 20],\n",
       " [1058, 'loss', 21],\n",
       " [1059, 'loss', 20],\n",
       " [1060, 'loss', 20],\n",
       " [1061, 'loss', 19],\n",
       " [1062, 'loss', 20],\n",
       " [1063, 'loss', 20],\n",
       " [1064, 'loss', 19],\n",
       " [1065, 'loss', 19],\n",
       " [1066, 'loss', 19],\n",
       " [1067, 'loss', 17],\n",
       " [1068, 'loss', 17],\n",
       " [1069, 'loss', 17],\n",
       " [1085, 'loss', 17],\n",
       " [1086, 'loss', 17],\n",
       " [1090, 'loss', 17],\n",
       " [1092, 'loss', 17],\n",
       " [1458, 'loss', 20],\n",
       " [1528, 'loss', 18],\n",
       " [1775, 'loss', 17],\n",
       " [1777, 'loss', 17],\n",
       " [1849, 'loss', 18],\n",
       " [2042, 'loss', 23],\n",
       " [2043, 'loss', 23],\n",
       " [2044, 'loss', 24],\n",
       " [2047, 'loss', 18],\n",
       " [2048, 'loss', 21],\n",
       " [2049, 'loss', 23],\n",
       " [2050, 'loss', 22],\n",
       " [2051, 'loss', 23],\n",
       " [2052, 'loss', 23],\n",
       " [2053, 'loss', 23],\n",
       " [2054, 'loss', 22],\n",
       " [2055, 'loss', 23],\n",
       " [2056, 'loss', 23],\n",
       " [2057, 'loss', 21],\n",
       " [2058, 'loss', 23],\n",
       " [2059, 'loss', 17],\n",
       " [2060, 'loss', 23],\n",
       " [2064, 'loss', 23],\n",
       " [2065, 'loss', 24],\n",
       " [2066, 'loss', 24],\n",
       " [2067, 'loss', 18],\n",
       " [2068, 'loss', 24],\n",
       " [2069, 'loss', 24],\n",
       " [2070, 'loss', 24],\n",
       " [2071, 'loss', 24],\n",
       " [2072, 'loss', 23],\n",
       " [2073, 'loss', 22],\n",
       " [2074, 'loss', 23],\n",
       " [2075, 'loss', 25],\n",
       " [2076, 'loss', 25],\n",
       " [2077, 'loss', 25],\n",
       " [2078, 'loss', 24],\n",
       " [2079, 'loss', 25],\n",
       " [2080, 'loss', 25],\n",
       " [2081, 'loss', 23],\n",
       " [2082, 'loss', 24],\n",
       " [2083, 'loss', 23],\n",
       " [2084, 'loss', 22],\n",
       " [2085, 'loss', 22],\n",
       " [2086, 'loss', 22],\n",
       " [2087, 'loss', 21],\n",
       " [2088, 'loss', 20],\n",
       " [2089, 'loss', 21],\n",
       " [2090, 'loss', 22],\n",
       " [2091, 'loss', 21],\n",
       " [2092, 'loss', 21],\n",
       " [2093, 'loss', 21],\n",
       " [2094, 'loss', 21],\n",
       " [2096, 'loss', 18],\n",
       " [2097, 'loss', 20],\n",
       " [2098, 'loss', 21],\n",
       " [2099, 'loss', 21],\n",
       " [2100, 'loss', 20],\n",
       " [2101, 'loss', 19],\n",
       " [2102, 'loss', 18],\n",
       " [2147, 'loss', 17],\n",
       " [2149, 'loss', 17],\n",
       " [2152, 'loss', 18],\n",
       " [2153, 'loss', 17],\n",
       " [2154, 'loss', 18],\n",
       " [2155, 'loss', 17],\n",
       " [2156, 'loss', 17],\n",
       " [2159, 'loss', 17],\n",
       " [2160, 'loss', 17],\n",
       " [2195, 'loss', 17],\n",
       " [2196, 'loss', 18],\n",
       " [2197, 'loss', 19],\n",
       " [2198, 'loss', 18],\n",
       " [2199, 'loss', 19],\n",
       " [2200, 'loss', 17],\n",
       " [2201, 'loss', 18],\n",
       " [2203, 'loss', 20],\n",
       " [2204, 'loss', 20],\n",
       " [2205, 'loss', 21],\n",
       " [2206, 'loss', 22],\n",
       " [2207, 'loss', 24],\n",
       " [2208, 'loss', 24],\n",
       " [2209, 'loss', 22],\n",
       " [2210, 'loss', 23],\n",
       " [2211, 'loss', 24],\n",
       " [2212, 'loss', 21],\n",
       " [2213, 'loss', 21],\n",
       " [2214, 'loss', 24],\n",
       " [2215, 'loss', 23],\n",
       " [2216, 'loss', 27],\n",
       " [2217, 'loss', 27],\n",
       " [2218, 'loss', 25],\n",
       " [2219, 'loss', 18],\n",
       " [2220, 'loss', 23],\n",
       " [2221, 'loss', 22],\n",
       " [2222, 'loss', 22],\n",
       " [2224, 'loss', 21],\n",
       " [2225, 'loss', 20],\n",
       " [2226, 'loss', 18],\n",
       " [2722, 'loss', 21],\n",
       " [2723, 'loss', 22]]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_values(her2_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[470, 'loss', 20],\n",
       " [471, 'loss', 23],\n",
       " [547, 'loss', 23],\n",
       " [761, 'loss', 27],\n",
       " [762, 'loss', 25],\n",
       " [815, 'loss', 19],\n",
       " [816, 'loss', 19],\n",
       " [817, 'loss', 19],\n",
       " [818, 'loss', 19],\n",
       " [819, 'loss', 19],\n",
       " [820, 'loss', 19],\n",
       " [832, 'loss', 19],\n",
       " [833, 'loss', 19],\n",
       " [834, 'loss', 19],\n",
       " [835, 'loss', 19],\n",
       " [836, 'loss', 19],\n",
       " [837, 'loss', 19],\n",
       " [838, 'loss', 19],\n",
       " [839, 'loss', 19],\n",
       " [840, 'loss', 19],\n",
       " [841, 'loss', 19],\n",
       " [845, 'loss', 19],\n",
       " [846, 'loss', 19],\n",
       " [847, 'loss', 19],\n",
       " [848, 'loss', 19],\n",
       " [849, 'loss', 19],\n",
       " [850, 'loss', 19],\n",
       " [851, 'loss', 19],\n",
       " [852, 'loss', 19],\n",
       " [853, 'loss', 20],\n",
       " [854, 'loss', 20],\n",
       " [855, 'loss', 20],\n",
       " [856, 'loss', 20],\n",
       " [857, 'loss', 20],\n",
       " [858, 'loss', 20],\n",
       " [859, 'loss', 19],\n",
       " [860, 'loss', 19],\n",
       " [861, 'loss', 19],\n",
       " [862, 'loss', 19],\n",
       " [863, 'loss', 19],\n",
       " [864, 'loss', 19],\n",
       " [865, 'loss', 19],\n",
       " [866, 'loss', 19],\n",
       " [867, 'loss', 19],\n",
       " [868, 'loss', 19],\n",
       " [869, 'loss', 19],\n",
       " [870, 'loss', 19],\n",
       " [871, 'loss', 20],\n",
       " [873, 'loss', 20],\n",
       " [874, 'loss', 20],\n",
       " [875, 'loss', 20],\n",
       " [883, 'loss', 19],\n",
       " [884, 'loss', 19],\n",
       " [885, 'loss', 19],\n",
       " [890, 'loss', 19],\n",
       " [892, 'loss', 19],\n",
       " [893, 'loss', 19],\n",
       " [894, 'loss', 19],\n",
       " [895, 'loss', 19],\n",
       " [896, 'loss', 20],\n",
       " [984, 'loss', 24],\n",
       " [1001, 'loss', 19],\n",
       " [1002, 'loss', 19],\n",
       " [1003, 'loss', 19],\n",
       " [1004, 'loss', 19],\n",
       " [1005, 'loss', 19],\n",
       " [1006, 'loss', 19],\n",
       " [1007, 'loss', 19],\n",
       " [1008, 'loss', 19],\n",
       " [1009, 'loss', 20],\n",
       " [1013, 'loss', 20],\n",
       " [1014, 'loss', 19],\n",
       " [1015, 'loss', 19],\n",
       " [1016, 'loss', 20],\n",
       " [1017, 'loss', 20],\n",
       " [1018, 'loss', 20],\n",
       " [1019, 'loss', 21],\n",
       " [1020, 'loss', 20],\n",
       " [1021, 'loss', 21],\n",
       " [1022, 'loss', 20],\n",
       " [1023, 'loss', 19],\n",
       " [1024, 'loss', 21],\n",
       " [1025, 'loss', 21],\n",
       " [1026, 'loss', 20],\n",
       " [1027, 'loss', 19],\n",
       " [1028, 'loss', 19],\n",
       " [1029, 'loss', 19],\n",
       " [1030, 'loss', 19],\n",
       " [1031, 'loss', 19],\n",
       " [1458, 'loss', 19],\n",
       " [1582, 'loss', 19],\n",
       " [1583, 'loss', 19],\n",
       " [1585, 'loss', 19],\n",
       " [1777, 'loss', 19],\n",
       " [1849, 'loss', 22],\n",
       " [1873, 'loss', 21],\n",
       " [2005, 'loss', 23],\n",
       " [2020, 'loss', 21],\n",
       " [2021, 'loss', 19],\n",
       " [2022, 'loss', 27],\n",
       " [2031, 'loss', 22],\n",
       " [2032, 'loss', 19],\n",
       " [2040, 'loss', 19],\n",
       " [2074, 'loss', 19],\n",
       " [2075, 'loss', 19],\n",
       " [2076, 'loss', 19],\n",
       " [2077, 'loss', 20],\n",
       " [2078, 'loss', 20],\n",
       " [2079, 'loss', 20],\n",
       " [2080, 'loss', 20],\n",
       " [2081, 'loss', 20],\n",
       " [2082, 'loss', 20],\n",
       " [2083, 'loss', 20],\n",
       " [2086, 'loss', 20],\n",
       " [2089, 'loss', 19],\n",
       " [2216, 'loss', 20],\n",
       " [2217, 'loss', 27],\n",
       " [2218, 'loss', 27],\n",
       " [2722, 'loss', 23],\n",
       " [2723, 'loss', 24],\n",
       " [2743, 'loss', 28]]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_values(hr_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[119, 'loss', 19],\n",
       " [470, 'loss', 17],\n",
       " [471, 'loss', 22],\n",
       " [547, 'loss', 23],\n",
       " [695, 'loss', 17],\n",
       " [761, 'loss', 26],\n",
       " [762, 'loss', 24],\n",
       " [984, 'loss', 17],\n",
       " [1006, 'loss', 18],\n",
       " [1007, 'loss', 18],\n",
       " [1008, 'loss', 18],\n",
       " [1009, 'loss', 17],\n",
       " [1013, 'loss', 18],\n",
       " [1016, 'loss', 18],\n",
       " [1017, 'loss', 18],\n",
       " [1018, 'loss', 18],\n",
       " [1019, 'loss', 18],\n",
       " [1020, 'loss', 18],\n",
       " [1021, 'loss', 18],\n",
       " [1022, 'loss', 18],\n",
       " [1023, 'loss', 18],\n",
       " [1024, 'loss', 18],\n",
       " [1025, 'loss', 18],\n",
       " [1026, 'loss', 17],\n",
       " [1028, 'loss', 17],\n",
       " [1031, 'loss', 18],\n",
       " [1032, 'loss', 18],\n",
       " [1033, 'loss', 18],\n",
       " [1036, 'loss', 17],\n",
       " [1037, 'loss', 18],\n",
       " [1038, 'loss', 18],\n",
       " [1039, 'loss', 18],\n",
       " [1040, 'loss', 18],\n",
       " [1041, 'loss', 17],\n",
       " [1042, 'loss', 17],\n",
       " [1410, 'loss', 18],\n",
       " [1458, 'loss', 17],\n",
       " [1528, 'loss', 19],\n",
       " [1775, 'loss', 18],\n",
       " [1776, 'loss', 17],\n",
       " [1777, 'loss', 19],\n",
       " [1849, 'loss', 21],\n",
       " [1870, 'loss', 18],\n",
       " [1873, 'loss', 19],\n",
       " [1902, 'loss', 18],\n",
       " [1903, 'loss', 18],\n",
       " [1907, 'loss', 17],\n",
       " [1908, 'loss', 17],\n",
       " [2216, 'loss', 17],\n",
       " [2217, 'loss', 23],\n",
       " [2218, 'loss', 23],\n",
       " [2723, 'loss', 17],\n",
       " [2743, 'loss', 17]]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_values(tn_array)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HER2: Human Epidermal growth factor Receptor 2: ERBB2, a known proto-oncogene, is located at the long arm of human chromosome 17 (17q12). \n",
    "\n",
    "PR: Progesterone Receptor: n humans, PR is encoded by a single PGR gene residing on chromosome 11q22, it has two isoforms, PR-A and PR-B, that differ in their molecular weight.\n",
    "\n",
    "ER: Estrogen Receptor: In humans, the two forms of the estrogen receptor are encoded by different genes, ESR1 and ESR2 on the 6 and 14 chromosome (6q25.1 and 14q23.2), respectively. \n",
    "\n",
    "HR+: Hormone Receptor Positive: 1)ER+ and HER2-, 2)PR+, and HER2-, 3)ER+, PR+ and HER2-\n",
    "HER2+\n",
    "Triple negative: ER-, PR- and HER2-"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
